pkg/nfd-worker/nfd-worker.go

```diff
@@ -172,6 +178,41 @@ func newDefaultConfig() *NFDConfig {
 	}
 }
 
+func (infiniteTicker *infiniteTicker) stopTicker() {
+	if infiniteTicker.initialized {
+		infiniteTicker.T.Stop()
+		infiniteTicker.initialized = false
+	}
+}
+
+func (infiniteTicker *infiniteTicker) startTicker(w *nfdWorker) {
+	// If the sleep interval is not a positive number, the ticker will act as if it was set to an infinite duration by not ticking.
+	if w.config.Core.SleepInterval.Duration > 0 {
+		infiniteTicker.T = time.NewTicker(w.config.Core.SleepInterval.Duration)
+		infiniteTicker.initialized = true
+	}
+}
```

Comment:
Thanks for the update @VillePihlava. Quite a bike-shedding on this PR, but somehow this latest version feels just a tad too complicated (in terms of lines-of-code) for such a simple thing 😇

How about using struct composition to wrap the Ticker type, with something like

```golang
type infiniteTicker struct {
        *time.Ticker
}

func (i *infiniteTicker) Reset(d time.Duration) {
        if d > 0 {
                i.Ticker.Reset(d)
        } else {
                i.Ticker.Stop()
        }
}
```

...then you'd initialize with smth like (you could specify a dedicated create function if this looks too bad):

```golang
labelTrigger := infiniteTicker{Ticker: time.NewTicker(1)} // just any non-zero duration here, we'll do the actual configuration below
labelTrigger.Reset(w.config.Core.SleepInterval.Duration)
```

use the channel directly:

```golang
case <-labelTrigger.C:
```
and simply use reset for re-configuring (instead of creating a new Ticker instance)

```golang
labelTrigger.Reset(w.config.Core.SleepInterval.Duration)
```

WDYT? Haven't tested but I think that should work out

Reply:
Yeah this makes sense, I made these changes now. Also tested manually that making changes to the sleepinterval in the config worked